subprojects/provider/src/test/kotlin/org/gradle/kotlin/dsl/resolver/KotlinBuildScriptModelRepositoryTest.kt

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.kotlin.dsl.resolver
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.async
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.channels.sendBlocking
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withTimeout
+import kotlinx.coroutines.withTimeoutOrNull
+
+import org.gradle.kotlin.dsl.tooling.models.EditorReport
+import org.gradle.kotlin.dsl.tooling.models.KotlinBuildScriptModel
+
+import org.hamcrest.CoreMatchers.nullValue
+import org.hamcrest.CoreMatchers.sameInstance
+import org.hamcrest.MatcherAssert.assertThat
+
+import org.junit.Test
+
+import java.io.File
+
+import java.util.concurrent.ArrayBlockingQueue
+import java.util.concurrent.TimeUnit
+
+import kotlin.coroutines.Continuation
+
+
+class KotlinBuildScriptModelRepositoryTest {
+
+    @Test
+    fun `cancels older requests before returning response to most recent request`() = runBlockingWithTimeout {
+
+        // given:
+        val projectDir = File("/tmp/project")
```

Comment:
:x: this should use a `TemporaryFolder` rule or `FolderBasedTest`

Reply:
I thought that would be adding unnecessary overhead for the files are never written to, they are just passed around as values. WDYT?

Reply:
Makes sense. What about capturing this more explicitly for the future reader then?